src/content/docs/en/guides/view-transitions.mdx

```diff
@@ -235,6 +235,34 @@ You can use this event to run code on every page navigation, or only once ever:
   }, { once: true });
 </script>
 ```
+
+#### `astro:beforeload`
+
+An event that fires immediately after the new page replaces the old page. You can listen to this event on the `document`.
+
+This event is useful to restore any state on the DOM that needs to transfer over to the new page. If you are implementing dark mode support, this event can be used to restore state onto the new page like so:
```

Comment:
Breaking up the generic “this is useful” and the specific example across paragraphs:

```suggestion
This event is useful to restore any state on the DOM that needs to transfer over to the new page.

For example, if you are implementing dark mode support, this event can be used to restore state across page loads:
```